File: game_engine.rb
class PiroEngine
  def initialize
    @games = {}
  end

  def create_game(game_name)
    @games[game_name] = 'initialized'
  end

  def start_game(game_name)
    if @games[game_name] == 'initialized'
      @games[game_name] = 'running'
    end
  end

  def stop_game(game_name)
    if @games[game_name] == 'running'
      @games[game_name] = 'stopped'
    end
  end

  def list_running_games
    @games.select { |_, status| status == 'running' }.keys
  end

  def list_initialized_games
    @games.select { |_, status| status == 'initialized' }.keys
  end
end

# Пример использования PiroEngine
engine = PiroEngine.new
engine.create_game('Game1')
engine.start_game('Game1')

running_games = engine.list_running_games
initialized_games = engine.list_initialized_games

puts 'Running Games:', running_games
puts 'Initialized Games:', initialized_games

File: game_engine.go
package main

import "fmt"

type PiroEngine struct {
    games map[string]string
}

func NewPiroEngine() *PiroEngine {
    return &PiroEngine{
        games: make(map[string]string),
    }
}

func (engine *PiroEngine) CreateGame(gameName string) {
    engine.games[gameName] = "initialized"
}

func (engine *PiroEngine) StartGame(gameName string) {
    if status, exists := engine.games[gameName]; exists {
        if status == "initialized" {
            engine.games[gameName] = "running"
        }
    }
}

func (engine *PiroEngine) StopGame(gameName string) {
    if status, exists := engine.games[gameName]; exists {
        if status == "running" {
            engine.games[gameName] = "stopped"
        }
    }
}

func (engine *PiroEngine) ListRunningGames() []string {
    runningGames := []string{}
    for gameName, status := range engine.games {
        if status == "running" {
            runningGames = append(runningGames, gameName)
        }
    }
    return runningGames
}

func (engine *PiroEngine) ListInitializedGames() []string {
    initializedGames := []string{}
    for gameName, status := range engine.games {
        if status == "initialized" {
            initializedGames = append(initializedGames, gameName)
        }
    }
    return initializedGames
}

func main() {
    engine := NewPiroEngine()
    engine.CreateGame("Game1")
    engine.StartGame("Game1")

    runningGames := engine.ListRunningGames()
    initializedGames := engine.ListInitializedGames()

    fmt.Println("Running Games:", runningGames)
    fmt.Println("Initialized Games:", initializedGames)
}
